module/actor/skill-handler.mjs

```diff
@@ -174,6 +175,48 @@ export function generateSkills(actor) {
     }
 }
 
+function extractSkillAttributes(actor) {
+    let skillAttributes = [];
+    if (!actor.system.changes) return skillAttributes;
+
+    for (let [_, changes] of Object.entries(actor.system.changes)) {
+        if (changes.key === "skillAttribute") {
+            skillAttributes.push(changes);
+        }
+    }
+    return skillAttributes;
+}
+
+function getModifiedSkillAttribute(skillAttributes, key) {
```

Comment:
it looks like this will select the first available attribute.  is that what you want?

Reply:
Other option would be to select them by prioritization/order, but in 99% of all cases it will be enough to select the first one, yes.
What do you think?

Reply:
can we have it select the highest modifier?